src/shims/unix/fs.rs

```diff
@@ -1448,16 +1454,41 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
 
                 let file_type = this.file_type_to_d_type(dir_entry.file_type())?;
 
-                this.write_int_fields_named(
-                    &[
-                        ("d_ino", ino.into()),
-                        ("d_seekoff", 0),
-                        ("d_reclen", 0),
-                        ("d_namlen", file_name_len.into()),
-                        ("d_type", file_type.into()),
-                    ],
-                    &entry_place,
-                )?;
+                if this.tcx.sess.target.os == "macos" {
```

Comment:
Please don't switch on the OS at all, instead just do the `if this.project_field_hasname(&entry_place, "d_off")` part.

Also add a comment explaining why we need to do this.